### PR TITLE
Added protection against recursive MCParticleManager::createPrimary()…

### PR DESCRIPTION
**Pull request by decibelcooper**
… from getting out of control

Added after observation of infinite or very large loop in createPrimary() for very large events from Pythia8.

### CODE REVIEW

**Comment by JeremyMcCormick** [comment]
Unless I'm missing something here this PR doesn't seem to do anything.  I don't see where you actually use the data structure to break out of the loop if a max is exceeded.

File: src/MCParticleManager.cc [hunk 1]
```diff
@@ -262,19 +262,23 @@ std::set<G4PrimaryParticle*> MCParticleManager::createPrimary(G4PrimaryParticle*
 #endif
     }
 
-    for (int i = 0; i < mcp->getDaughters().size(); i++) {
+    if (_recursionBlockSet.count(mcp) == 0) {
```
Comment (decibelcooper) [inline src/MCParticleManager.cc:265]: This is where exclusion occurs.  This breaks recursion if the daughters were already scanned.

Comment (decibelcooper) [reply src/MCParticleManager.cc:265]: There is no hard max, it is a fix for recursion loops.

**Comment by JeremyMcCormick** [comment]
Okay I missed that....is this the best place to check for this?  Perhaps it
should happen upstream where the lcio particle list is being constructed.

**Comment by JeremyMcCormick** [comment]
This is out of date now and I have not been able to test it.  I have also never seen any infinite loop issues with any of the stdhep files we have used.  I am going to close it.

**Pull request closed without merging**